src/metabase/api/dataset.clj

```diff
@@ -59,7 +59,10 @@
   "Execute a query and retrieve the results in the usual format."
   [:as {{:keys [database], query-type :type, :as query} :body}]
   {database (s/maybe s/Int)}
-  (run-query-async (update query :middleware assoc :js-int-to-string? true)))
+  (run-query-async (update-in query [:middleware :js-int-to-string?] (fn [existing-value]
```

Comment:
`(update-in query [...] (fnil identity true))` might be clearer and more concise